values.go

```diff
@@ -132,6 +135,61 @@ func (value *Number) Equal(that Value) bool {
 	return value.value == typed.value && value.typ.scale == typed.typ.scale
 }
 
+func (value *Number) GreaterThan(that Value) bool {
+	typed, ok := that.(*Number)
+	if !ok {
+		return false
+	}
+	if value.typ.scale > typed.typ.scale {
+		return value.value > typed.scaleUp(value.typ.scale)
+	}
+	if value.typ.scale < typed.typ.scale {
+		return value.scaleUp(typed.typ.scale) > typed.value
+	}
+	return value.value > typed.value
+}
+
+// for usual number equality, we require that value and scale are equal;
+// for gt/gte/lt/lte comparisons, however, we recognize that scale doesn't need
+// to be equal because we'll often be bounds checking (e.g., 5.999999 <= 6.0)
+func (value *Number) ScaledEqual(that Value) bool {
```

Comment:
Maybe `NumericEqual` and make it clear that `that` can only be a number. Here we can ask the caller to check that the value is a number,